### PR TITLE
Update error message for allowed characters in aggregation names

### PR DESCRIPTION
**Pull request by ielatif**
Closing #41567

### CODE REVIEW

**Comment by ielatif** [comment]
PR for #41567

**Comment by ielatif** [comment]
Hi @cbuescher I'm new contributor to elasticsearch and this is my first PR. Can you help with the next steps please?

File: server/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java [hunk 1]
```diff
@@ -75,7 +75,7 @@
             final String aggregationName = parser.currentName();
             if (!validAggMatcher.reset(aggregationName).matches()) {
                 throw new ParsingException(parser.getTokenLocation(), "Invalid aggregation name [" + aggregationName
-                        + "]. Aggregation names must be alpha-numeric and can only contain '_' and '-'");
+                        + "]. Aggregation names must be any character except '[', ']', and '>'");
```
Comment (polyfractal) [inline server/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java]: Could we make the message say: `Aggregation names can contain any character except '[', ']', and '>'` ?

Reads a little clearner than "must be any character except...".

**Review by polyfractal — CHANGES_REQUESTED** [review]
Thanks for the PR @ielatif!  I left a comment about the text.  

I'm also kicking off a CI build.  There may be some tests that check specific exception text which might need updating too.  The CI run should let us know if anything breaks.

@elasticmachine test this please

File: server/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java (hunk 1 above)
Comment (ielatif) [reply server/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java]: Thx @polyfractal.
I agree, it sounds more cleaner. I will change it and check for test failures

Comment (ielatif) [reply server/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java]: I updated the message and run ./gradlew chech on the server module but dont get any test failure.

Comment (polyfractal) [reply server/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java]: Oops, looks like the new commit erased the CI results.  I'll kick off another build! :)

**Comment by polyfractal** [comment]
@elasticmachine test this please

**Comment by ielatif** [comment]
@polyfractal CI build fails because it try to load openjdk-12.0.1_linux-x64_bin with wrong extension xml.
I created this issue #41688

**Comment by polyfractal** [comment]
Could you merge in master?  I believe there were some issues related to Oracle changing their download locations and certificates not matching, which required some code changes in the packaging tests.  I think it was resolved in: https://github.com/elastic/elasticsearch/commit/5fa17aaf8ad1b303bf81611c44e5b51c97dab18e

**Comment by ielatif** [comment]
@polyfractal Done. Could you plz ask @elasticmachine to run tests?

**Comment by polyfractal** [comment]
Thanks!

@elasticmachine test this please

**Comment by ielatif** [comment]
@polyfractal Thanks! It worked :+1:

**Comment by ielatif** [comment]
@polyfractal is it ok for you?

**Review by polyfractal — APPROVED** [review]
LGTM!  Sorry for the delay, had a busy day yesterday. I'll get this merged to the appropriate branches today :)  Thanks @ielatif

**Comment by ielatif** [comment]
You're welcome. I am glad to contribute.

**Comment by polyfractal** [comment]
Merged, thanks again @ielatif!

**Pull request merged**